Source/WebKit/UIProcess/Extensions/WebExtension.h

```diff
@@ -192,79 +202,78 @@ class WebExtension : public API::ObjectImpl<API::Object::Type::WebExtension>, pu
     bool operator==(const WebExtension& other) const { return (this == &other); }
 
     bool manifestParsedSuccessfully();
-    NSDictionary *manifest();
+    RefPtr<JSON::Value> *manifest();
     Ref<API::Data> serializeManifest();
 
     double manifestVersion();
     bool supportsManifestVersion(double version) { ASSERT(version > 2); return manifestVersion() >= version; }
 
     Ref<API::Data> serializeLocalization();
 
+#if PLATFORM(COCOA)
     SecStaticCodeRef bundleStaticCode() const;
     NSData *bundleHash() const;
 
-#if PLATFORM(MAC)
     bool validateResourceData(NSURL *, NSData *, NSError **);
+
+    UTType *resourceTypeForPath(StringView);
+    NSLocale *defaultLocale();
 #endif
 
     bool isWebAccessibleResource(const URL& resourceURL, const URL& pageURL);
```

Comment:
Am I missing it or are some of these missing implementations?

Reply:
resourceTypeForPath is only used in the Cocoa imageForPath impl, just to do something special if the image is SVG.

defaultLocale() isn't implemented because there's a lot more that's needed with localisation support (including porting the shared WKWebExtensionLocalization classes), and I figured it's better to leave that alone until the full localisation support is done (so as not to regress the Cocoa build).